### PR TITLE
Swc 648 - Java File uploader

### PR DESCRIPTION
**Pull request by dburdick**

### CODE REVIEW

File: client/fileUploadClient/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	<name>fileUploadClient</name>
+
+	<dependencies>
+
+		<dependency>
+			<groupId>org.sagebionetworks</groupId>
+			<artifactId>synapseJavaClient</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.pivot</groupId>
```
Comment (dburdick) [inline client/fileUploadClient/pom.xml:25]: add full dependency to root pom, remove version from here

File: client/fileUploadClient/pom.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			<artifactId>pivot-core</artifactId>
+			<version>2.0</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.pivot</groupId>
+			<artifactId>pivot-wtk-terra</artifactId>
+			<version>2.0</version>
+		</dependency>
+		<dependency>
+			<groupId>jmimemagic</groupId>
```
Comment (dburdick) [inline client/fileUploadClient/pom.xml]: Look into Java 7 api to see if mime type (guessMimeType?) is newer than jmimemagic

see ContentTypeUtil

File: client/fileUploadClient/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+				<directory>src/main/java</directory>
+			</resource>
+			<resource>
+				<!-- We also want the resources directory added -->
+				<directory>src/main/resources</directory>
+			</resource>
+		</resources>
+		<plugins>
+			<plugin>
+				<artifactId>maven-assembly-plugin</artifactId>
```
Comment (dburdick) [inline client/fileUploadClient/pom.xml:59]: look into signing

File: client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/App.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	private static Boolean isUpdate;
+	
+	public static void main(String[] args) {
+		logger.info("starting app");
+		if(args != null && args.length >= 2) {
+			for(String arg : args) {
+				if(arg.startsWith("--sessionToken=")) sessionToken = arg.replace("--sessionToken=", "");
+				if(arg.startsWith("--entityId=")) entityId = arg.replace("--entityId=", "");
+			}
+			logger.info("sessiontoken: "+ sessionToken + ", parentId: "+ entityId + ", isUpdate: "+ isUpdate);
```
Comment (dburdick) [inline client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/App.java]: remove isUpdate

File: client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/App.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+		}
+		DesktopApplicationContext.main(App.class, args);				
+	}
+
+	@Override
+    public void startup(Display display, Map<String, String> properties) throws Exception {
+		if(sessionToken == null || entityId == null ) {
+			showUsage(display);
+			return;
```
Comment (dburdick) [inline client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/App.java]: throw IllegalArgumentException

File: client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/FileUploaderViewImpl.java [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+    }
+
+	@Override
+	public void alert(String message) {
+		Alert.alert(message, this);
+	}
+
+	@Override
+	public void updateFileStatus() {
+		fileTableView.repaint();
```
Comment (dburdick) [inline client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/FileUploaderViewImpl.java]: check that repaint can be called from a non event dispatch thread

File: client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/FileUploaderViewImpl.java [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+            }
+
+            @Override
+            public DropAction drop(Component component, Manifest dragContent,
+                int supportedDropActions, int x, int y, DropAction userDropAction) {
+                DropAction dropAction = null;
+
+                if (dragContent.containsFileList()) {
+                    try {
+                        presenter.addFilesForUpload(dragContent.getFileList().getList());
```
Comment (dburdick) [inline client/fileUploadClient/src/main/java/org/sagebionetworks/client/fileuploader/FileUploaderViewImpl.java:197]: alert user and cancel the drop if in single file mode and more than one file is dropped.

**Comment by john-hill** [comment]
Looks good!

**Pull request merged**